sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -469,6 +475,30 @@ public static string CanonicalizeResourcePathV2(Uri endpoint, string resourcePat
             return canonicalizedResourcePath;
         }
 
+        /// <summary>
+        /// This method returns the full canonicalized resource path which includes the resource path from the request and the resource path from the endpoint.
+        /// This is just a helper method to be used internally.
+        /// </summary>
+        /// <param name="encodedSegments">If double encoded, encoded segments contains the encoded segments from the resource path from the request</param>
+        /// <param name="resourcePathFromEndpoint">This is the resource path that comes from the endpoint itself and not the request</param>
+        /// <returns>The full canonicalized resource path that contains the single encoded canonicalized resource path from the endpoint and the double encoded
+        /// resource path from the request</returns>
+        private static string GetFullCanonicalizedResourcePath(IEnumerable<string> encodedSegments, string resourcePathFromEndpoint)
+        {
+            // Most of the time resourcePathFromEndpoint will just be "/", meaning it is essentially empty, so check to see if the length is > 1.
+            if (resourcePathFromEndpoint.Length > 1)
+            {
+                // Since we are resolving the configured endpoint here, we don't want to subsitute any key/value pairs from path resources.
+                IEnumerable<string> absolutePathSegment = AWSSDKUtils.SplitResourcePathIntoSegments(resourcePathFromEndpoint, null);
+                // If encodedSegments' first value is empty, that means the resourcePath was empty, so no need to concatenate it since that will produce an extra "/" character at the end.
+                IEnumerable<string> fullPath = string.IsNullOrEmpty(encodedSegments.First()) ? absolutePathSegment : absolutePathSegment.Concat(encodedSegments);
```

Comment:
Can we use `FirstOrDefault` instead of `First` to protect ourselves in case `encodedSegments` is ever an empty collection.